test/test_torch.py

```diff
@@ -4120,6 +4120,63 @@ def logcumsumexp(a, axis):
                 'expected scalar_type Double but found Float'):
             torch.logcumsumexp(b, axis, out=inplace_out)
 
+    # RngNormal not implemented for type f16 for XLA
+    @dtypes(*torch.testing.get_all_dtypes(include_half=False))
+    @dtypesIfCPU(*torch.testing.get_all_dtypes())
+    @dtypesIfCUDA(*torch.testing.get_all_dtypes())
+    def test_diff(self, device, dtype):
+        def _test_diff_numpy(t, dims=None):
+            for dim in dims if dims else range(t.dim()):
+                actual = torch.diff(t, dim=dim, prepend=t, append=t)
+
+                if t.dtype == torch.bfloat16:
+                    np_t = t.to(dtype=torch.float, device="cpu").numpy()
+                else:
+                    np_t = t.cpu().numpy()
+
+                # test when both prepend and append are tensors
+                expected = torch.from_numpy(np.diff(np_t, axis=dim, prepend=np_t, append=np_t))
+                self.assertEqual(actual, expected.to(t.dtype))
+
+        shapes = (
+            (1,),
+            (1, 5),
+            (3, 5),
+            (1, 5, 1),
+            (2, 3, 5))
+
+        for shape in shapes:
+            contig = _make_tensor(shape, dtype, device)
+            _test_diff_numpy(contig)
+
+            non_contig = torch.empty(shape + (2, 2), dtype=dtype)[..., 0]
```

Comment:
Use make_tensor instead of _make_tensor:

https://github.com/pytorch/pytorch/blob/4a2aa0f5f1572d8a9caca9496669f8f52cfa1522/torch/testing/_internal/common_utils.py#L1482

Both here on 4152 and on 4149.

Reply:
I've replaced _make_tensor with make_tensor on line 4149, but I don't think we can use make_tensor to produce torch.empty for line 4152.